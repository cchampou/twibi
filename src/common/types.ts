export interface Stream {
  id: string;
  user_id: string;
  user_login: string;
  user_name: string;
  game_id: string;
  game_name: string;
  type: string;
  title: string;
  viewer_count: number;
  started_at: string;
  language: string;
  thumbnail_url: string;
  tag_ids: string;
}

export interface Channel {
  broadcaster_id: string;
  broadcaster_name: string;
  game_name: string;
  game_id: string;
  broadcaster_language: string;
  title: string;
  delay: number;
}

export interface User {
  broadcaster_type: string;
  description: string;
  display_name: string;
  id: string;
  login: string;
  offline_image_url: string;
  profile_image_url: string;
  type: string;
  view_count: number;
  email: string;
  created_at: string;
}

export interface CommandType {
  command: string;
  response: string;
}
