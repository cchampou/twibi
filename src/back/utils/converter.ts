import { MessageEmbedOptions } from 'discord.js';
import { Stream } from '../../common/types';

// eslint-disable-next-line import/prefer-default-export
export const streamToDiscordEmbed: (streamData: Stream) => MessageEmbedOptions =
  (streamData) => ({
    title: streamData.title,
    color: 0xff0000,
    fields: [
      { name: 'ðŸ‘‡', value: 'https://www.twitch.tv/carorockwell', inline: true },
    ],
    author: {
      name: 'Carorockwell',
      iconURL:
        'https://images-ext-1.discordapp.net/external/Ekh9K5IqwNP313ZdvmYFf57f_FIdh0JDgruQBxuq1lw/https/static-cdn.jtvnw.net/jtv_user_pictures/d0fc61fb-b675-4ce7-b37a-63a1a22c5926-profile_image-300x300.png',
      url: 'https://www.twitch.tv/carorockwell',
    },
    url: 'https://www.twitch.tv/carorockwell',
    image: {
      proxy_url:
        'https://media.discordapp.net/attachments/828259641171378198/847016522668965918/IMG_20210518_215042_152.jpg?width=386&height=686',
    },
    thumbnail: {
      proxy_url:
        'https://images-ext-1.discordapp.net/external/Ekh9K5IqwNP313ZdvmYFf57f_FIdh0JDgruQBxuq1lw/https/static-cdn.jtvnw.net/jtv_user_pictures/d0fc61fb-b675-4ce7-b37a-63a1a22c5926-profile_image-300x300.png',
    },
    description: 'Viens nous rejoindre sur Twitch !',
  });
