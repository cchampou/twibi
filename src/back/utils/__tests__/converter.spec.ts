import { streamToDiscordEmbed } from '../converter';
import { Stream } from '../../../common/types';

const sampleData: Stream = {
  id: '123',
  user_id: '123456',
  user_login: 'bg',
  user_name: 'BG',
  game_id: 'sample',
  game_name: 'sample',
  type: 'sample',
  title: 'Titre du stream',
  viewer_count: 5,
  started_at: 'a date',
  language: 'sample',
  thumbnail_url: 'sample',
  tag_ids: 'sample',
};

describe('streamToDiscordEmbed', () => {
  it('should match the previous format', () => {
    expect(streamToDiscordEmbed(sampleData)).toMatchSnapshot();
  });
});
