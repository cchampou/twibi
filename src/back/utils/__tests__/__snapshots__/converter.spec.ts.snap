// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`streamToDiscordEmbed should match the previous format 1`] = `
Object {
  "author": Object {
    "iconURL": "https://images-ext-1.discordapp.net/external/Ekh9K5IqwNP313ZdvmYFf57f_FIdh0JDgruQBxuq1lw/https/static-cdn.jtvnw.net/jtv_user_pictures/d0fc61fb-b675-4ce7-b37a-63a1a22c5926-profile_image-300x300.png",
    "name": "Carorockwell",
    "url": "https://www.twitch.tv/carorockwell",
  },
  "color": 16711680,
  "description": "Viens nous rejoindre sur Twitch !",
  "fields": Array [
    Object {
      "inline": true,
      "name": "ðŸ‘‡",
      "value": "https://www.twitch.tv/carorockwell",
    },
  ],
  "image": Object {
    "proxy_url": "https://media.discordapp.net/attachments/828259641171378198/847016522668965918/IMG_20210518_215042_152.jpg?width=386&height=686",
  },
  "thumbnail": Object {
    "proxy_url": "https://images-ext-1.discordapp.net/external/Ekh9K5IqwNP313ZdvmYFf57f_FIdh0JDgruQBxuq1lw/https/static-cdn.jtvnw.net/jtv_user_pictures/d0fc61fb-b675-4ce7-b37a-63a1a22c5926-profile_image-300x300.png",
  },
  "title": "Titre du stream",
  "url": "https://www.twitch.tv/carorockwell",
}
`;
